Docs: Update grafana configuration documentation (#75717)
grafana/grafana: docs/sources/setup-grafana/configure-grafana/_index.md

### auto_assign_org_role

> The role that new users will be assigned for the main organization (if the
> `auto_assign_org` setting is set to true). Defaults to `Viewer`, other valid
> options are `Admin`, `Editor` and `None`. e.g.:

Reviewer: ```suggestion
The `auto_assign_org` setting determines the default role assigned to new users in the main organization. The available options are `Viewer` (default), `Admin`, `Editor`, and `None`.
```

Author: Hey @chri2547, thanks for the review. 

The configuration variable's name is `auto_assign_org_role` actually and it is only relevant when `auto_assign_org` is set to true. I've taken your suggestion into account and made the changes. Could you please take a look?
